.flake8

```diff
@@ -1,7 +1,8 @@
 [flake8]
-max-line-length = 95
+max-line-length = 120
```

Comment:
Let's update this [in `ruff.toml`](https://github.com/sphinx-doc/sphinx/blob/392358d4de91c560a833074ddf9a4c6fc2b07a62/.ruff.toml#L2) (and anywhere else?) too.

Reply:
no, this is just to stop flake8 complaining in places ruff does do not; ruff allows leniency in line-length for lines with comments, flake8 does not

Reply:
> In fact: I think `ruff` will read it from `flake8`'s config if it can. So we should _remove_ it from `ruff.toml`, I think.

No, I may be talking rubbish unfortunately.  I was distracted by the fact that `ruff config line-length` _mentions_ `120`, which is, by chance, what we're updating `flake8` to use here.  But it doesn't confirm that `ruff` is reading that value.

Reply:
i would actually suggest suppressing the lint in flake8 entirely, rather than just making the line length limit greater

Reply:
yep i've confirmed ruff still uses 95

Reply:
> i would actually suggest suppressing the lint in flake8 entirely, rather than just making the line length limit greater

Only for comments, ideally?

Reply:
[`max-doc-length`](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-max-doc-length) could achieve that I think, if so.

Reply:
> how do you do that?

add `E501` to the 'ignore' list in the flake8 config file

> Only for comments, ideally?

ruff ignores comments when calculating line length, flake8 does not.
ruff allows longer lines in a few select cases:

> Overlong lines can hurt readability. [PEP 8](https://peps.python.org/pep-0008/#maximum-line-length), for example, recommends limiting lines to 79 characters. By default, this rule enforces a limit of 88 characters for compatibility with Black, though that limit is configurable via the [line-length](https://docs.astral.sh/ruff/settings/#line-length) setting.
> 
> In the interest of pragmatism, this rule makes a few exceptions when determining whether a line is overlong. Namely, it:
> 
> Ignores lines that consist of a single "word" (i.e., without any whitespace between its characters).
> Ignores lines that end with a URL, as long as the URL starts before the line-length threshold.
> Ignores line that end with a pragma comment (e.g., # type: ignore or # noqa), as long as the pragma comment starts before the line-length threshold. That is, a line will not be flagged as overlong if a pragma comment causes it to exceed the line length. (This behavior aligns with that of the Ruff formatter.)

Reply:
> add E501 to the 'ignore' list in the flake8 config file

done 👍